core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -278,6 +285,7 @@ protected void startMasters(boolean isLeader) {
    */
   protected void stopMasters() {
     try {
+      mWorkerConfigChecker.reset();
```

Comment:
Should this be moved to `startMasters`? @aaudiber

Reply:
By @aaudiber :Actually I agree with Lu that it's better to call init during start instead of during stop. It's better to initialize closer to when we need it so that there's less time for something to interfere. When we call stop() there could still be outstanding RPC threads trying to register workers and modifying the server config checker

@gpang I prefer to change the reset() to startMasters() part to make sure the leader master have empty conf map when it get leadership, PTAL, thanks